src/data_gradients/feature_extractors/common/image_duplicates.py

```diff
@@ -0,0 +1,264 @@
+        dup_clique_heads = list(dups.keys())
+
+        # ITERATE THROUGH THE CLIQUE 'HEADS' (I.E SOME MEMBER REPRESENTING THE DUPLICATE CLIQUE)
+        for i in range(len(dup_clique_heads)):
+            dup_key = dup_clique_heads[i]
+
+            # IF THIS CLIQUE HEAD WAS ALREADY ADDED IN PREVIOUS ITERATIONS, SKIP IT TO AVOID ADDING THE SAME CLIQUE
+            # TWICE (SINCE find_duplicates WILL RETURN THE DUPLICATES IN EVERY MEMBER'S ENTRY)
+
+            if self._is_in_dup_clique(dup_key, train_dups) or self._is_in_dup_clique(dup_key, valid_dups) or self._is_in_dup_clique(dup_key, intersection_dups):
+                continue
+
+            # CREATE A 'CLIQUE` FROM THE MEMBER AND IT'S DUPLICATES:
+            # IMAGE_X_PATH: [DUPLICATE_OF_IMAGE_X_PATH, DUPLICATE_2_OF_IMAGE_X_PATH...] ->
+            # [IMAGE_X_PATH, DUPLICATE_OF_IMAGE_X_PATH, DUPLICATE_2_OF_IMAGE_X_PATH...]
+            dup_clique = self._make_dup_clique(dup_key, dups)
+
+            # IF THE CLIQUE HAS AT LEAST 2 PATHS IN THE TRAIN/VALIDATION DIR -
+            # ADD IT TO train_dups/valid_dups AFTER FILTERING PATHS OUTSIDE THE TRAIN/VALIDATION DIR.
+            if self._is_train_dup(dup_clique):
+                train_dups.append([d for d in dup_clique if d.startswith(self.train_image_dir)])
+            if self._is_valid_dup(dup_clique):
+                valid_dups.append([d for d in dup_clique if d.startswith(self.valid_image_dir)])
+
+            # IF THE CLIQUE HAS IT LEAST ONE PATH IN EACH - ADD IT TO intersection_dups
+            if self._is_intersection_dup(dup_clique):
+                intersection_dups.append(dup_clique)
+
+        self.train_dups, self.valid_dups, self.intersection_dups = train_dups, valid_dups, intersection_dups
+        self.train_dups_appearences = self._count_dir_dup_appearences(self.train_dups, self.train_image_dir)
+        self.validation_dups_appearences = self._count_dir_dup_appearences(self.valid_dups, self.valid_image_dir)
+        self.intersection_train_appearnces = self._count_dir_dup_appearences(self.intersection_dups, self.train_image_dir)
+        self.intersection_val_appearnces = self._count_dir_dup_appearences(self.intersection_dups, self.valid_image_dir)
+
+    @staticmethod
+    def _is_in_dup_clique(sample: str, dup_clique_list: List[List[str]]) -> bool:
+        """
+        Whether sample is already in some duplicated clique in dup_clique_list.
+
+        :param sample: str, an image path
+        :param dup_clique_list: List[List[str]], list of duplicate image path cliques (lists).
+        """
+        return any([sample in d for d in dup_clique_list])
+
+    @staticmethod
+    def _make_dup_clique(dup_key, dups):
+        dup_clique = [dup_key] + dups[dup_key]
+        return dup_clique
+
+    def _is_train_dup(self, dup_clique: List[str]) -> bool:
+        """
+        Whether the dup_clique contains duplicate image paths from train_image_dir.
+
+        :param dup_clique: str, list of duplicated image paths
```

Comment:
`:param dup_clique: str, list of duplicated image paths` -> `:param dup_clique: List of duplicated image paths`

Reply:
fixed in latest commit.